### PR TITLE
Electron application support

### PR DESCRIPTION
**Pull request by brendanny**
- added configuration files to support electron-forge and electron-vite

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,22 +2,31 @@
   "name": "music-box-interactive-client",
   "private": true,
   "version": "2.3.6",
+  "main": "dist/main/main.js",
```
Comment (brendanny) [inline package.json:5]: Will this affect vite?

File: package.json [hunk 2]
```diff
@@ -2,22 +2,31 @@
...
   "dependencies": {
     "axios": "^1.6.2",
     "better-react-mathjax": "^2.0.3",
     "bootstrap": "^5.3.2",
     "d3": "^7.8.5",
+    "electron-squirrel-startup": "^1.0.0",
+    "is-electron": "^2.2.2",
     "open-iconic": "^1.1.1",
     "prop-types": "^15.8.1",
+    "pyodide": "^0.25.0",
```
Comment (brendanny) [inline package.json:29]: Needed to check python version. is there a way to enforce a conditional dependency so its not necessary to build the web app?

File: src/renderer/index.html
Comment (brendanny) [inline src/renderer/index.html:1]: `npm run dev` and `npm run build` work on my local machine with out errors. However `npm run dev` starts the local dev server but I can't open in my browser on my machine

File: vite.config.js
Comment (brendanny) [inline vite.config.js:1]: still need to double check that this change is necessary

File: src/renderer/index.html
Comment (brendanny) [reply src/renderer/index.html:1]: @K20shores are there some vite config changes that I need to add after moving the entry file?

File: src/renderer/vite.config.js [hunk 3]
```diff
@@ -8,15 +8,15 @@ export default defineConfig({
   },
   define: {
     "process.env.VITE_APP_VERSION": JSON.stringify(
-      require("./package.json").version
+      require("../../package.json").version
```
Comment (brendanny) [inline src/renderer/vite.config.js:11]: do we need to make a separate package.json or is referring to the root one fine?

File: vite.config.js [hunk 4]
```diff
@@ -8,7 +8,15 @@ export default defineConfig({
...
     ),
   },
+  build: {
+    rollupOptions: {
+      input: {
+        main: "src/renderer/index.html", // Entry for the main page
+      },
+    },
+    outDir: "dist",
+  },
```
[the remark this answers is not in the document]
Comment (brendanny) [reply vite.config.js]: it builds commented out so i've removed it for now in https://github.com/NCAR/music-box-interactive-client/pull/248/commits/bc039ba1f3ace6bb015b2a451f9e8aace9efcfad

**Review by K20shores — APPROVED** [review]

**Pull request merged**